diff --git a/frameworks/native/avdemuxer/avdemuxer_impl.cpp b/frameworks/native/avdemuxer/avdemuxer_impl.cpp
index ea6755380..b64ff82d8 100644
--- a/frameworks/native/avdemuxer/avdemuxer_impl.cpp
+++ b/frameworks/native/avdemuxer/avdemuxer_impl.cpp
@@ -223,27 +223,27 @@ int32_t AVDemuxerImpl::GetGopLayerInfo(uint32_t gopId, GopLayerInfo &gopLayerInf
     return StatusToAVCodecServiceErrCode(demuxerEngine_->GetGopLayerInfo(gopId, gopLayerInfo));
 }
 
-int32_t AVDemuxerImpl::GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-    int64_t presentationTimeUs, uint32_t &frameIndex)
+int32_t AVDemuxerImpl::GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+    const uint64_t relativePresentationTimeUs, uint32_t &index)
 {
     AVCODEC_SYNC_TRACE;
-    AVCODEC_LOGD("GetFrameIndexByPresentationTimeUs");
+    AVCODEC_LOGD("GetIndexByRelativePresentationTimeUs");
     CHECK_AND_RETURN_RET_LOG(demuxerEngine_ != nullptr, AVCS_ERR_INVALID_OPERATION,
         "Demuxer engine does not exist");
-    int32_t ret = StatusToAVCodecServiceErrCode(demuxerEngine_->GetFrameIndexByPresentationTimeUs(trackIndex,
-        presentationTimeUs, frameIndex));
+    int32_t ret = StatusToAVCodecServiceErrCode(demuxerEngine_->GetIndexByRelativePresentationTimeUs(trackIndex,
+        relativePresentationTimeUs, index));
     return ret;
 }
 
-int32_t AVDemuxerImpl::GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-    uint32_t frameIndex, int64_t &presentationTimeUs)
+int32_t AVDemuxerImpl::GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+    const uint32_t index, uint64_t &relativePresentationTimeUs)
 {
     AVCODEC_SYNC_TRACE;
-    AVCODEC_LOGD("GetPresentationTimeUsByFrameIndex");
+    AVCODEC_LOGD("GetRelativePresentationTimeUsByIndex");
     CHECK_AND_RETURN_RET_LOG(demuxerEngine_ != nullptr, AVCS_ERR_INVALID_OPERATION,
         "Demuxer engine does not exist");
-    int32_t ret = StatusToAVCodecServiceErrCode(demuxerEngine_->GetPresentationTimeUsByFrameIndex(trackIndex,
-        frameIndex, presentationTimeUs));
+    int32_t ret = StatusToAVCodecServiceErrCode(demuxerEngine_->GetRelativePresentationTimeUsByIndex(trackIndex,
+        index, relativePresentationTimeUs));
     return ret;
 }
 } // namespace MediaAVCodec
diff --git a/frameworks/native/avdemuxer/avdemuxer_impl.h b/frameworks/native/avdemuxer/avdemuxer_impl.h
index eb7c19eeb..120ee2db1 100644
--- a/frameworks/native/avdemuxer/avdemuxer_impl.h
+++ b/frameworks/native/avdemuxer/avdemuxer_impl.h
@@ -44,10 +44,10 @@ public:
     int32_t StartReferenceParser(int64_t startTimeMs) override;
     int32_t GetFrameLayerInfo(std::shared_ptr<AVBuffer> videoSample, FrameLayerInfo &frameLayerInfo) override;
     int32_t GetGopLayerInfo(uint32_t gopId, GopLayerInfo &gopLayerInfo) override;
-    int32_t GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-        int64_t presentationTimeUs, uint32_t &frameIndex) override;
-    int32_t GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-        uint32_t frameIndex, int64_t &presentationTimeUs) override;
+    int32_t GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+        const uint64_t relativePresentationTimeUs, uint32_t &index) override;
+    int32_t GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+        const uint32_t index, uint64_t &relativePresentationTimeUs) override;
 
 private:
     std::shared_ptr<MediaDemuxer> demuxerEngine_ = nullptr;
diff --git a/interfaces/inner_api/native/avdemuxer.h b/interfaces/inner_api/native/avdemuxer.h
index 55addf6e7..6705f10a2 100644
--- a/interfaces/inner_api/native/avdemuxer.h
+++ b/interfaces/inner_api/native/avdemuxer.h
@@ -106,10 +106,10 @@ public:
     virtual int32_t GetFrameLayerInfo(std::shared_ptr<AVBuffer> videoSample, FrameLayerInfo &frameLayerInfo) = 0;
     virtual int32_t GetGopLayerInfo(uint32_t gopId, GopLayerInfo &gopLayerInfo) = 0;
 
-    virtual int32_t GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-        int64_t presentationTimeUs, uint32_t &frameIndex) = 0;
-    virtual int32_t GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-        uint32_t frameIndex, int64_t &presentationTimeUs) = 0;
+    virtual int32_t GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+        const uint64_t relativePresentationTimeUs, uint32_t &index) = 0;
+    virtual int32_t GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+        const uint32_t index, uint64_t &relativePresentationTimeUs) = 0;
 };
 
 class __attribute__((visibility("default"))) AVDemuxerFactory {
diff --git a/interfaces/inner_api/native/media_demuxer.h b/interfaces/inner_api/native/media_demuxer.h
index b17231d21..a8f085b20 100755
--- a/interfaces/inner_api/native/media_demuxer.h
+++ b/interfaces/inner_api/native/media_demuxer.h
@@ -117,8 +117,10 @@ public:
     void RegisterVideoStreamReadyCallback(const std::shared_ptr<VideoStreamReadyCallback> &callback);
     void DeregisterVideoStreamReadyCallback();
 
-    Status GetFrameIndexByPresentationTimeUs(uint32_t trackIndex, int64_t presentationTimeUs, uint32_t &frameIndex);
-    Status GetPresentationTimeUsByFrameIndex(uint32_t trackIndex, uint32_t frameIndex, int64_t &presentationTimeUs);
+    Status GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+        const uint64_t relativePresentationTimeUs, uint32_t &index);
+    Status GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+        const uint32_t index, uint64_t &relativePresentationTimeUs);
     Status ResumeDemuxerReadLoop();
     Status PauseDemuxerReadLoop();
     void SetCacheLimit(uint32_t limitSize);
diff --git a/interfaces/plugin/demuxer_plugin.h b/interfaces/plugin/demuxer_plugin.h
index 6ddba3707..5fdc08196 100644
--- a/interfaces/plugin/demuxer_plugin.h
+++ b/interfaces/plugin/demuxer_plugin.h
@@ -160,11 +160,11 @@ struct DemuxerPlugin : public PluginBase {
         return Status::OK;
     }
 
-    virtual Status GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-        int64_t presentationTimeUs, uint32_t &frameIndex) = 0;
+    virtual Status GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+        const uint64_t relativePresentationTimeUs, uint32_t &index) = 0;
 
-    virtual Status GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-        uint32_t frameIndex, int64_t &presentationTimeUs) = 0;
+    virtual Status GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+        const uint32_t index, uint64_t &relativePresentationTimeUs) = 0;
 
     virtual void SetCacheLimit(uint32_t limitSize) = 0;
 };
diff --git a/services/media_engine/modules/demuxer/media_demuxer.cpp b/services/media_engine/modules/demuxer/media_demuxer.cpp
index da7d268be..f516b24db 100755
--- a/services/media_engine/modules/demuxer/media_demuxer.cpp
+++ b/services/media_engine/modules/demuxer/media_demuxer.cpp
@@ -2081,36 +2081,36 @@ bool MediaDemuxer::IsRenderNextVideoFrameSupported()
         !isDataSrcLiveStream;
 }
 
-Status MediaDemuxer::GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-    int64_t presentationTimeUs, uint32_t &frameIndex)
+Status MediaDemuxer::GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+    const uint64_t relativePresentationTimeUs, uint32_t &index)
 {
-    MEDIA_LOG_D("GetFrameIndexByPresentationTimeUs");
+    MEDIA_LOG_D("GetIndexByRelativePresentationTimeUs");
     FALSE_RETURN_V_MSG_E(demuxerPluginManager_ != nullptr, Status::ERROR_NULL_POINTER,
-        "GetFrameIndexByPresentationTimeUs failed due to demuxerPluginManager_ is nullptr.");
+        "GetIndexByRelativePresentationTimeUs failed due to demuxerPluginManager_ is nullptr.");
     std::shared_ptr<Plugins::DemuxerPlugin> pluginTemp = GetCurFFmpegPlugin();
     FALSE_RETURN_V_MSG_E(pluginTemp != nullptr, Status::ERROR_NULL_POINTER,
-        "GetFrameIndexByPresentationTimeUs failed due to get demuxer plugin failed.");
+        "GetIndexByRelativePresentationTimeUs failed due to get demuxer plugin failed.");
 
-    Status ret = pluginTemp->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
+    Status ret = pluginTemp->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
     if (ret != Status::OK) {
-        MEDIA_LOG_E("MediaDemuxer GetFrameIndexByPresentationTimeUs failed");
+        MEDIA_LOG_E("MediaDemuxer GetIndexByRelativePresentationTimeUs failed");
     }
     return ret;
 }
 
-Status MediaDemuxer::GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-    uint32_t frameIndex, int64_t &presentationTimeUs)
+Status MediaDemuxer::GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+    const uint32_t index, uint64_t &relativePresentationTimeUs)
 {
-    MEDIA_LOG_D("GetPresentationTimeUsByFrameIndex");
+    MEDIA_LOG_D("GetRelativePresentationTimeUsByIndex");
     FALSE_RETURN_V_MSG_E(demuxerPluginManager_ != nullptr, Status::ERROR_NULL_POINTER,
-        "GetPresentationTimeUsByFrameIndex failed due to demuxerPluginManager_ is nullptr.");
+        "GetRelativePresentationTimeUsByIndex failed due to demuxerPluginManager_ is nullptr.");
     std::shared_ptr<Plugins::DemuxerPlugin> pluginTemp = GetCurFFmpegPlugin();
     FALSE_RETURN_V_MSG_E(pluginTemp != nullptr, Status::ERROR_NULL_POINTER,
-        "GetPresentationTimeUsByFrameIndex failed due to get demuxer plugin failed.");
+        "GetRelativePresentationTimeUsByIndex failed due to get demuxer plugin failed.");
 
-    Status ret = pluginTemp->GetPresentationTimeUsByFrameIndex(trackIndex, frameIndex, presentationTimeUs);
+    Status ret = pluginTemp->GetRelativePresentationTimeUsByIndex(trackIndex, index, relativePresentationTimeUs);
     if (ret != Status::OK) {
-        MEDIA_LOG_E("MediaDemuxer GetPresentationTimeUsByFrameIndex failed");
+        MEDIA_LOG_E("MediaDemuxer GetRelativePresentationTimeUsByIndex failed");
     }
     return ret;
 }
diff --git a/services/media_engine/plugins/ffmpeg_adapter/BUILD.gn b/services/media_engine/plugins/ffmpeg_adapter/BUILD.gn
index 61ce73064..67c669364 100644
--- a/services/media_engine/plugins/ffmpeg_adapter/BUILD.gn
+++ b/services/media_engine/plugins/ffmpeg_adapter/BUILD.gn
@@ -85,6 +85,8 @@ ohos_shared_library("media_plugin_FFmpegDemuxer") {
 
   sanitize = av_codec_sanitize
 
+  defines = ["OHOS_EXPAND_MP4_INFO"]
+
   configs = [
     ":ffmpeg_adapter_config",
     "$av_codec_root_dir/services/dfx:av_codec_service_log_dfx_public_config",
diff --git a/services/media_engine/plugins/ffmpeg_adapter/demuxer/ffmpeg_demuxer_plugin.cpp b/services/media_engine/plugins/ffmpeg_adapter/demuxer/ffmpeg_demuxer_plugin.cpp
index decf5fd5d..7aaddc3da 100644
--- a/services/media_engine/plugins/ffmpeg_adapter/demuxer/ffmpeg_demuxer_plugin.cpp
+++ b/services/media_engine/plugins/ffmpeg_adapter/demuxer/ffmpeg_demuxer_plugin.cpp
@@ -91,11 +91,6 @@ inline int64_t AvTime2Us(int64_t hTime)
     return hTime / AV_CODEC_USECOND;
 }
 
-inline int64_t AvUs2Time(int64_t hTime)
-{
-    return hTime * AV_CODEC_USECOND;
-}
-
 static const std::map<SeekMode, int32_t>  g_seekModeToFFmpegSeekFlags = {
     { SeekMode::SEEK_PREVIOUS_SYNC, AVSEEK_FLAG_BACKWARD },
     { SeekMode::SEEK_NEXT_SYNC, AVSEEK_FLAG_FRAME },
@@ -1903,72 +1898,166 @@ bool FFmpegDemuxerPlugin::CanDropHevcPkt(const AVPacket& pkt)
         nalUnitType == 6 || nalUnitType == 8; // 6: RADL_N, 8: RASL_N
 }
 
-Status FFmpegDemuxerPlugin::GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-    int64_t presentationTimeUs, uint32_t &frameIndex)
+Status FFmpegDemuxerPlugin::GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+    const uint64_t relativePresentationTimeUs, uint32_t &index)
 {
+    RelativePTSToIndexPosition_ = 0; // init RelativePTSToIndexPosition_
+    IndexToRelativePTSMaxHeap_ = std::priority_queue<int64_t>(); // init IndexToRelativePTSMaxHeap_
+
     FALSE_RETURN_V_MSG_E(formatContext_ != nullptr, Status::ERROR_NULL_POINTER,
-        "GetFrameIndexByPresentationTimeUs failed due to formatContext_ is nullptr.");
+        "GetIndexByRelativePresentationTimeUs failed due to formatContext_ is nullptr.");
 
     FALSE_RETURN_V_MSG_E(trackIndex < formatContext_->nb_streams, Status::ERROR_INVALID_DATA,
-        "GetFrameIndexByPresentationTimeUs failed due to trackIndex is out of range.");
+        "GetIndexByRelativePresentationTimeUs failed due to trackIndex is out of range.");
 
     FALSE_RETURN_V_MSG_E(FFmpegFormatHelper::GetFileTypeByName(*formatContext_) == FileType::MP4,
-        Status::ERROR_MISMATCHED_TYPE, "GetFrameIndexByPresentationTimeUs failed due to fileType is not MP4.");
-
-    auto avStream = formatContext_->streams[trackIndex];
-    FALSE_RETURN_V_MSG_E(avStream != nullptr, Status::ERROR_NULL_POINTER,
-        "GetFrameIndexByPresentationTimeUs failed due to avStream is nullptr.");
+        Status::ERROR_MISMATCHED_TYPE, "GetIndexByRelativePresentationTimeUs failed due to fileType is not MP4.");
+    
+    GetpresentationTimeUsFromFfmpegMOV(Get_FIRST_PTS, trackIndex,
+        static_cast<int64_t>(relativePresentationTimeUs), index);
 
-    int64_t pts = presentationTimeUs;
-    presentationTimeUs = AvUs2Time(ConvertTimeToFFmpeg(presentationTimeUs, avStream->time_base));
+    int64_t absolutePTS = static_cast<int64_t>(relativePresentationTimeUs) + AbsolutePTSIndexZero_;
 
-    int index = av_index_search_timestamp(avStream, presentationTimeUs, AVSEEK_FLAG_ANY);
-    FALSE_RETURN_V_MSG_E(index >= 0, Status::ERROR_INVALID_DATA,
-        "GetFrameIndexByPresentationTimeUs failed due to index is invalid data.");
-    
-    int64_t convertPts;
-    Status ret = GetPresentationTimeUsByFrameIndex(trackIndex, static_cast<uint32_t>(index), convertPts);
-    FALSE_RETURN_V_MSG_E(ret == Status::OK, ret,
-        "GetFrameIndexByPresentationTimeUs failed due to GetPresentationTimeUsByFrameIndex is failed.");
+    GetpresentationTimeUsFromFfmpegMOV(RELATIVEPTS_TO_INDEX, trackIndex,
+        absolutePTS, index);
     
-    FALSE_RETURN_V_MSG_E(convertPts == pts, Status::ERROR_INVALID_DATA,
-        "GetFrameIndexByPresentationTimeUs failed due to presentationTimeUs don't correspond to frameIndex.");
+    if (absolutePTS < RelativePTSToIndexPTSMin_ || absolutePTS > RelativePTSToIndexPTSMax_) {
+        return Status::ERROR_INVALID_DATA;
+    }
+
+    if (RelativePTSToIndexLeftDiff_ == 0 || RelativePTSToIndexRightDiff_ == 0) {
+        index = RelativePTSToIndexPosition_;
+        return Status::OK;
+    }
 
-    frameIndex = static_cast<uint32_t>(index);
+    index = RelativePTSToIndexLeftDiff_ < RelativePTSToIndexRightDiff_ ?
+        RelativePTSToIndexPosition_ - 1 : RelativePTSToIndexPosition_;
+    
     return Status::OK;
 }
 
-Status FFmpegDemuxerPlugin::GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-    uint32_t frameIndex, int64_t &presentationTimeUs)
+Status FFmpegDemuxerPlugin::GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+    const uint32_t index, uint64_t &relativePresentationTimeUs)
 {
+    RelativePTSToIndexPosition_ = 0; // init RelativePTSToIndexPosition_
+    IndexToRelativePTSMaxHeap_ = std::priority_queue<int64_t>(); // init IndexToRelativePTSMaxHeap_
+
     FALSE_RETURN_V_MSG_E(formatContext_ != nullptr, Status::ERROR_NULL_POINTER,
-        "GetPresentationTimeUsByFrameIndex failed due to formatContext_ is nullptr.");
+        "GetRelativePresentationTimeUsByIndex failed due to formatContext_ is nullptr.");
 
     FALSE_RETURN_V_MSG_E(trackIndex < formatContext_->nb_streams, Status::ERROR_INVALID_DATA,
-        "GetPresentationTimeUsByFrameIndex failed due to trackIndex is out of range.");
+        "GetRelativePresentationTimeUsByIndex failed due to trackIndex is out of range.");
 
     FALSE_RETURN_V_MSG_E(FFmpegFormatHelper::GetFileTypeByName(*formatContext_) == FileType::MP4,
-        Status::ERROR_MISMATCHED_TYPE, "GetPresentationTimeUsByFrameIndex failed due to fileType is not MP4.");
+        Status::ERROR_MISMATCHED_TYPE, "GetRelativePresentationTimeUsByIndex failed due to fileType is not MP4.");
+    
+    GetpresentationTimeUsFromFfmpegMOV(Get_FIRST_PTS, trackIndex,
+        static_cast<int64_t>(relativePresentationTimeUs), index);
+    GetpresentationTimeUsFromFfmpegMOV(INDEX_TO_RELATIVEPTS, trackIndex,
+        static_cast<int64_t>(relativePresentationTimeUs), index);
+    if (index + 1 > IndexToRelativePTSFrameCount_) {
+        return Status::ERROR_INVALID_DATA;
+    }
 
+    relativePresentationTimeUs = static_cast<uint64_t>(IndexToRelativePTSMaxHeap_.top()
+        - AbsolutePTSIndexZero_);
+
+    return Status::OK;
+}
+
+Status FFmpegDemuxerPlugin::GetpresentationTimeUsFromFfmpegMOV(IndexAndPTSConvertMode mode, 
+    uint32_t trackIndex, int64_t absolutePTS, uint32_t index)
+{
     auto avStream = formatContext_->streams[trackIndex];
     FALSE_RETURN_V_MSG_E(avStream != nullptr, Status::ERROR_NULL_POINTER,
-        "GetPresentationTimeUsByFrameIndex failed due to avStream is nullptr.");
-    if (avStream->start_time == AV_NOPTS_VALUE || ioContext_.dataSource->IsDash()) {
-        avStream->start_time = 0;
+        "GetpresentationTimeUsFromFfmpegMOV failed due to avStream is nullptr.");
+
+    struct AVCodecMOVStts *stts = avStream->stts_data_head->next;
+    struct AVCodecMOVCtts *ctts = avStream->ctts_data_head;
+    int32_t ctts_cur_num = 0; //init ctts_cur_num
+    if (ctts->next != NULL) {
+        ctts = ctts->next;
+        ctts_cur_num = ctts->count;
+    } else {
+        ctts->next = ctts;
     }
+    int64_t pts = 0; // init pts
+    int64_t dts = 0; // init dts
+    int32_t stts_cur_num = stts->count;
 
-    const AVIndexEntry *entry = avformat_index_get_entry(avStream, frameIndex);
-    FALSE_RETURN_V_MSG_E(entry != nullptr, Status::ERROR_NULL_POINTER, "Invalid frameIndex");
+    while (true) {
+        if (ctts_cur_num == 0 && ctts->next != ctts) {
+            ctts = ctts->next;
+            ctts_cur_num = static_cast<int32_t>(ctts->count);
+        }
+        if (ctts->next != ctts) {
+            ctts_cur_num--;
+        }
+        pts = (dts + (ctts->next == ctts ? 0 : static_cast<int64_t>(ctts->duration))) * 1000 * 1000 /
+            static_cast<int64_t>(avStream->time_scale);
+        
+        switch (mode) {
+            case Get_FIRST_PTS:
+                AbsolutePTSIndexZero_ = pts < AbsolutePTSIndexZero_ ? pts : AbsolutePTSIndexZero_;
+                break;
+            case INDEX_TO_RELATIVEPTS:
+                IndexToRelativePTSProcess(pts, index);
+                break;
+            case RELATIVEPTS_TO_INDEX:
+                RelativePTSProcessToIndex(pts, absolutePTS);
+                break;
+            default:
+                MEDIA_LOG_E("wrong GetpresentationTimeUsFromFfmpegMOV mode");
+                break;
+        }
 
-    if (avStream->codecpar->codec_type == AVMEDIA_TYPE_VIDEO) {
-        int64_t inputPts = ConvertPts(entry->timestamp, avStream->start_time);
-        presentationTimeUs = AvTime2Us(ConvertTimeFromFFmpeg(inputPts, avStream->time_base));
-    } else {
-        presentationTimeUs = AvTime2Us(ConvertTimeFromFFmpeg(entry->timestamp, avStream->time_base));
+        stts_cur_num--;
+        if (stts->next == NULL && stts_cur_num ==0 &&
+            (ctts->next == NULL | ctts->next == ctts) && ctts_cur_num == 0) {
+                break;
+            }
+        dts += static_cast<int64_t>(stts->duration);
+        if (stts_cur_num == 0) {
+            stts = stts->next;
+            stts_cur_num = static_cast<int32_t>(stts->count);
+        }
     }
     return Status::OK;
 }
 
+void FFmpegDemuxerPlugin::IndexToRelativePTSProcess(int64_t pts, uint32_t index)
+{
+    if (IndexToRelativePTSMaxHeap_.size() < index + 1 ) {
+        IndexToRelativePTSMaxHeap_.push(pts);
+    } else {
+        if (pts < IndexToRelativePTSMaxHeap_.top()) {
+            IndexToRelativePTSMaxHeap_.pop();
+            IndexToRelativePTSMaxHeap_.push(pts);
+        }
+    }
+    IndexToRelativePTSFrameCount_++;
+}
+
+void FFmpegDemuxerPlugin::RelativePTSProcessToIndex(int64_t pts, int64_t AbsolutePTS)
+{
+    if (RelativePTSToIndexPTSMin_ > pts) {
+        RelativePTSToIndexPTSMin_ = pts;
+    }
+    if (RelativePTSToIndexPTSMax_ < pts) {
+        RelativePTSToIndexPTSMax_ = pts;
+    }
+    RelativePTSToIndexTempDiff_ = abs(pts - AbsolutePTS);
+    if (pts < AbsolutePTS && RelativePTSToIndexTempDiff_ < RelativePTSToIndexLeftDiff_) {
+        RelativePTSToIndexLeftDiff_ = RelativePTSToIndexTempDiff_;
+    }
+    if (pts >= AbsolutePTS && RelativePTSToIndexTempDiff_ < RelativePTSToIndexLeftDiff_) {
+        RelativePTSToIndexRightDiff_ = RelativePTSToIndexTempDiff_;
+    }
+    if (pts < AbsolutePTS) {
+        RelativePTSToIndexPosition_++;
+    }
+}
+
 Status FFmpegDemuxerPlugin::CheckCacheDataLimit(uint32_t trackId)
 {
     auto cacheDataSize = cacheQueue_.GetCacheDataSize(trackId);
diff --git a/services/media_engine/plugins/ffmpeg_adapter/demuxer/ffmpeg_demuxer_plugin.h b/services/media_engine/plugins/ffmpeg_adapter/demuxer/ffmpeg_demuxer_plugin.h
index eec9e8c57..7693b2f21 100644
--- a/services/media_engine/plugins/ffmpeg_adapter/demuxer/ffmpeg_demuxer_plugin.h
+++ b/services/media_engine/plugins/ffmpeg_adapter/demuxer/ffmpeg_demuxer_plugin.h
@@ -20,6 +20,7 @@
 #include <vector>
 #include <thread>
 #include <map>
+#include <queue>
 #include <shared_mutex>
 #include <list>
 #include "buffer/avbuffer.h"
@@ -72,10 +73,10 @@ public:
     Status GetGopLayerInfo(uint32_t gopId, GopLayerInfo &gopLayerInfo) override;
     Status GetIFramePos(std::vector<uint32_t> &IFramePos) override;
     Status Dts2FrameId(int64_t dts, uint32_t &frameId, bool offset = true) override;
-    Status GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-        int64_t presentationTimeUs, uint32_t &frameIndex) override;
-    Status GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-        uint32_t frameIndex, int64_t &presentationTimeUs) override;
+    Status GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+        const uint64_t relativePresentationTimeUs, uint32_t &index) override;
+    Status GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+        const uint32_t index, uint64_t &relativePresentationTimeUs) override;
     void SetCacheLimit(uint32_t limitSize) override;
 
 private:
@@ -85,6 +86,11 @@ private:
         DUMP_AVPACKET_OUTPUT = 0b010,
         DUMP_AVBUFFER_OUTPUT = 0b100,
     };
+    enum IndexAndPTSConvertMode : unsigned int {
+        Get_FIRST_PTS,
+        INDEX_TO_RELATIVEPTS,
+        RELATIVEPTS_TO_INDEX,
+    };
     struct IOContext {
         std::shared_ptr<DataSource> dataSource {nullptr};
         int64_t offset {0};
@@ -144,6 +150,21 @@ private:
     void WebvttMP4EOSProcess(AVPacket *vttPkt);
     Status CheckCacheDataLimit(uint32_t trackId);
 
+    Status GetpresentationTimeUsFromFfmpegMOV(IndexAndPTSConvertMode mode, 
+        uint32_t trackIndex, int64_t absolutePTS, uint32_t index);
+    void IndexToRelativePTSProcess(int64_t pts, uint32_t index);
+    void RelativePTSProcessToIndex(int64_t pts, int64_t AbsolutePTS);
+    int64_t AbsolutePTSIndexZero_ = INT64_MAX;
+    std::priority_queue<int64_t> IndexToRelativePTSMaxHeap_;
+    uint32_t IndexToRelativePTSFrameCount_ = 0;
+    uint32_t RelativePTSToIndexPosition_ = 0;
+    int64_t RelativePTSToIndexPTSMin_ = INT64_MAX;
+    int64_t RelativePTSToIndexPTSMax_ = INT64_MIN;
+    int64_t RelativePTSToIndexRightDiff_ = INT64_MAX;
+    int64_t RelativePTSToIndexLeftDiff_ = INT64_MAX;
+    int64_t RelativePTSToIndexTempDiff_ = INT64_MAX;
+
+
     std::mutex mutex_ {};
     std::shared_mutex sharedMutex_;
     std::unordered_map<uint32_t, std::shared_ptr<std::mutex>> trackMtx_;
diff --git a/test/moduletest/demuxer/src/InnerAPI/inner_demuxer_sample.cpp b/test/moduletest/demuxer/src/InnerAPI/inner_demuxer_sample.cpp
index 10b77c7d0..380cd5bd4 100644
--- a/test/moduletest/demuxer/src/InnerAPI/inner_demuxer_sample.cpp
+++ b/test/moduletest/demuxer/src/InnerAPI/inner_demuxer_sample.cpp
@@ -167,35 +167,35 @@ int32_t InnerDemuxerSample::CheckIndexFromPts()
 
 void InnerDemuxerSample::CheckLoopForPts(int32_t i)
 {
-    int64_t presentationTimeUs = 0;
+    uint64_t relativePresentationTimeUs = 0;
     int32_t num = 999;
     indexForPts = 0;
     if (i == videoTrackIdx) {
         for (const auto &pair : videoIndexPtsMap) {
-            retForPts = demuxer_->GetPresentationTimeUsByFrameIndex(i, indexForPts, presentationTimeUs);
-            cout << "video GetPresentationTimeUsByFrameIndex ret:" << retForPts << endl;
+            retForPts = demuxer_->GetRelativePresentationTimeUsByIndex(i, indexForPts, relativePresentationTimeUs);
+            cout << "video GetRelativePresentationTimeUsByIndex ret:" << retForPts << endl;
             if (retForPts != 0) {
-                cout << "GetPresentationTimeUsByFrameIndex fail ret:" << retForPts << endl;
+                cout << "GetRelativePresentationTimeUsByIndex fail ret:" << retForPts << endl;
                 break;
             }
-            if (presentationTimeUs != pair.second) {
+            if (relativePresentationTimeUs != pair.second) {
                 retForPts = num;
-                cout << "video != pair.second  pts:" << presentationTimeUs << "pair.second:"<< pair.second << endl;
+                cout << "video != pair.second  pts:" << relativePresentationTimeUs << "pair.second:"<< pair.second << endl;
                 break;
             }
             indexForPts ++;
         }
     } else {
         for (const auto &pair : audioIndexPtsMap) {
-            retForPts = demuxer_->GetPresentationTimeUsByFrameIndex(i, indexForPts, presentationTimeUs);
-            cout << "audio GetPresentationTimeUsByFrameIndex ret:" << retForPts << endl;
+            retForPts = demuxer_->GetRelativePresentationTimeUsByIndex(i, indexForPts, relativePresentationTimeUs);
+            cout << "audio GetRelativePresentationTimeUsByIndex ret:" << retForPts << endl;
             if (retForPts != 0) {
-                cout << "GetPresentationTimeUsByFrameIndex fail ret:" << retForPts << endl;
+                cout << "GetRelativePresentationTimeUsByIndex fail ret:" << retForPts << endl;
                 break;
             }
-            if (presentationTimeUs != pair.second) {
+            if (relativePresentationTimeUs != pair.second) {
                 retForPts = num;
-                cout << "audio != pair.second pts:" << presentationTimeUs << "pair.second:"<< pair.second << endl;
+                cout << "audio != pair.second pts:" << relativePresentationTimeUs << "pair.second:"<< pair.second << endl;
                 break;
             }
             indexForPts ++;
@@ -208,9 +208,9 @@ void InnerDemuxerSample::CheckLoopForIndex(int32_t i)
     int32_t num = 999;
     if (i == videoTrackIdx) {
         for (const auto &pair : videoIndexPtsMap) {
-            retForIndex = demuxer_->GetFrameIndexByPresentationTimeUs(i, pair.second, index);
+            retForIndex = demuxer_->GetIndexByRelativePresentationTimeUs(i, pair.second, index);
             if (retForIndex != 0) {
-                cout << "video GetFrameIndexByPresentationTimeUs fail ret:" << retForIndex << endl;
+                cout << "video GetIndexByRelativePresentationTimeUs fail ret:" << retForIndex << endl;
                 break;
             }
             if (index != pair.first) {
@@ -221,9 +221,9 @@ void InnerDemuxerSample::CheckLoopForIndex(int32_t i)
         }
     } else {
         for (const auto &pair : audioIndexPtsMap) {
-            retForIndex = demuxer_->GetFrameIndexByPresentationTimeUs(i, pair.second, index);
+            retForIndex = demuxer_->GetIndexByRelativePresentationTimeUs(i, pair.second, index);
             if (retForIndex != 0) {
-                cout << "audio GetFrameIndexByPresentationTimeUs fail ret:" << retForIndex << endl;
+                cout << "audio GetIndexByRelativePresentationTimeUs fail ret:" << retForIndex << endl;
                 break;
             }
             if (index != pair.first) {
diff --git a/test/nativedemo/avdemuxer/avdemuxer_demo_runner.cpp b/test/nativedemo/avdemuxer/avdemuxer_demo_runner.cpp
index 4e17555cb..de139dd83 100644
--- a/test/nativedemo/avdemuxer/avdemuxer_demo_runner.cpp
+++ b/test/nativedemo/avdemuxer/avdemuxer_demo_runner.cpp
@@ -21,6 +21,7 @@
 #include <thread>
 #include <sys/stat.h>
 #include <fstream>
+#include <chrono>
 
 #include "avcodec_common.h"
 #include "buffer/avsharedmemorybase.h"
@@ -245,15 +246,31 @@ static void RunDrmNativeDemuxer(const std::string &filePath, const std::string &
 
 static void ConvertPtsFrameIndexDemo(std::shared_ptr<InnerDemuxerDemo> innerDemuxerDemo)
 {
-    uint32_t trackIndex = 1;
-    int64_t presentationTimeUs = 100000;    // pts 100000
-    uint32_t frameIndex = 0;
-    innerDemuxerDemo->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
-    printf("GetFrameIndexByPresentationTimeUs, frameIndex = %d\n", frameIndex);
-    presentationTimeUs = 0;
-    frameIndex = 100;    // frameIndex 100
-    innerDemuxerDemo->GetPresentationTimeUsByFrameIndex(trackIndex, frameIndex, presentationTimeUs);
-    printf("GetPresentationTimeUsByFrameIndex, presentationTimeUs = %" PRId64 "\n", presentationTimeUs);
+    uint32_t trackIndex = 0;
+    uint64_t relativePresentationTimeUs = 0;    // pts 0
+    uint32_t index = 0;
+
+    using clock = std::chrono::high_resolution_clock;
+    auto start = clock::now();
+    auto end = clock::now();
+    std::chrono::duration<double> elapsed = end - start;
+
+    for (uint32_t i = 0; i < 10 ; ++i) {
+        index = i;
+        start = clock::now();
+        innerDemuxerDemo->GetRelativePresentationTimeUsByIndex(trackIndex, index, relativePresentationTimeUs);
+        end = clock::now();
+        elapsed = end - start;
+        printf("GetRelativePresentationTimeUsByIndex, relativePresentationTimeUs = %" PRId64 "\n", relativePresentationTimeUs); 
+        std::cout << "Function took " << elapsed.count() << " seconds to run.\n";
+
+        start = clock::now();
+        innerDemuxerDemo->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
+        end = clock::now();
+        elapsed = end - start;
+        printf("GetIndexByRelativePresentationTimeUs, index = %d\n", index);
+        std::cout << "Function took " << elapsed.count() << " seconds to run.\n";
+    }
 }
 
 static void RunInnerSourceDemuxer(const std::string &filePath, const std::string &fileMode)
diff --git a/test/nativedemo/avdemuxer/inner_demo/inner_demuxer_demo.cpp b/test/nativedemo/avdemuxer/inner_demo/inner_demuxer_demo.cpp
index fd70d56ec..1b4d1e823 100644
--- a/test/nativedemo/avdemuxer/inner_demo/inner_demuxer_demo.cpp
+++ b/test/nativedemo/avdemuxer/inner_demo/inner_demuxer_demo.cpp
@@ -157,22 +157,22 @@ int32_t InnerDemuxerDemo::SeekToTime(int64_t millisecond, Media::SeekMode mode)
     return ret;
 }
 
-int32_t InnerDemuxerDemo::GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-    int64_t presentationTimeUs, uint32_t &frameIndex)
+int32_t InnerDemuxerDemo::GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+    const uint64_t relativePresentationTimeUs, uint32_t &index)
 {
-    int32_t ret = demuxer_->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
+    int32_t ret = demuxer_->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
     if (ret != 0) {
-        printf("GetFrameIndexByPresentationTimeUs is failed\n");
+        printf("GetIndexByRelativePresentationTimeUs is failed\n");
     }
     return ret;
 }
 
-int32_t InnerDemuxerDemo::GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-    uint32_t frameIndex, int64_t &presentationTimeUs)
+int32_t InnerDemuxerDemo::GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+    const uint32_t index, uint64_t &relativePresentationTimeUs)
 {
-    int32_t ret = demuxer_->GetPresentationTimeUsByFrameIndex(trackIndex, frameIndex, presentationTimeUs);
+    int32_t ret = demuxer_->GetRelativePresentationTimeUsByIndex(trackIndex, index, relativePresentationTimeUs);
     if (ret != 0) {
-        printf("GetPresentationTimeUsByFrameIndex is failed\n");
+        printf("GetRelativePresentationTimeUsByIndex is failed\n");
     }
     return ret;
 }
diff --git a/test/nativedemo/avdemuxer/inner_demo/inner_demuxer_demo.h b/test/nativedemo/avdemuxer/inner_demo/inner_demuxer_demo.h
index a72fde5a4..b1d594e7d 100644
--- a/test/nativedemo/avdemuxer/inner_demo/inner_demuxer_demo.h
+++ b/test/nativedemo/avdemuxer/inner_demo/inner_demuxer_demo.h
@@ -39,8 +39,10 @@ public:
     int32_t ReadSample(uint32_t trackIndex, std::shared_ptr<AVSharedMemory> mem,
                         AVCodecBufferInfo &bufInfo, uint32_t &bufferFlag);
     int32_t SeekToTime(int64_t millisecond, Media::SeekMode mode);
-    int32_t GetFrameIndexByPresentationTimeUs(uint32_t trackIndex, int64_t presentationTimeUs, uint32_t &frameIndex);
-    int32_t GetPresentationTimeUsByFrameIndex(uint32_t trackIndex, uint32_t frameIndex, int64_t &presentationTimeUs);
+    int32_t GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+        const uint64_t relativePresentationTimeUs, uint32_t &index);
+    int32_t GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+        const uint32_t index, uint64_t &relativePresentationTimeUs);
     bool isEOS(std::map<uint32_t, bool>& countFlag);
     int32_t StartReferenceParser(int64_t startTimeMs);
     int32_t GetFrameLayerInfo(std::shared_ptr<AVBuffer> videoSample, FrameLayerInfo &frameLayerInfo);
diff --git a/test/unittest/demuxer_test/capi/demuxer_capi_buffer_mock.cpp b/test/unittest/demuxer_test/capi/demuxer_capi_buffer_mock.cpp
index 493103ee8..3c033a064 100644
--- a/test/unittest/demuxer_test/capi/demuxer_capi_buffer_mock.cpp
+++ b/test/unittest/demuxer_test/capi/demuxer_capi_buffer_mock.cpp
@@ -118,14 +118,14 @@ int32_t DemuxerCapiMock::GetMediaKeySystemInfo()
     return AV_ERR_OK;
 }
 
-int32_t DemuxerCapiMock::GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-    int64_t presentationTimeUs, uint32_t &frameIndex)
+int32_t DemuxerCapiMock::GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+    const uint64_t relativePresentationTimeUs, uint32_t &index)
 {
     return AV_ERR_OK;
 }
 
-int32_t DemuxerCapiMock::GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-    uint32_t frameIndex, int64_t &presentationTimeUs)
+int32_t DemuxerCapiMock::GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+    const uint32_t index, uint64_t &relativePresentationTimeUs)
 {
     return AV_ERR_OK;
 }
diff --git a/test/unittest/demuxer_test/capi/demuxer_capi_mock.cpp b/test/unittest/demuxer_test/capi/demuxer_capi_mock.cpp
index 26b0e0cf0..f39f5a283 100644
--- a/test/unittest/demuxer_test/capi/demuxer_capi_mock.cpp
+++ b/test/unittest/demuxer_test/capi/demuxer_capi_mock.cpp
@@ -171,14 +171,14 @@ int32_t DemuxerCapiMock::GetMediaKeySystemInfo()
     return AV_ERR_OK;
 }
 
-int32_t DemuxerCapiMock::GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-    int64_t presentationTimeUs, uint32_t &frameIndex)
+int32_t DemuxerCapiMock::GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+    const uint64_t relativePresentationTimeUs, uint32_t &index)
 {
     return AV_ERR_OK;
 }
 
-int32_t DemuxerCapiMock::GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-    uint32_t frameIndex, int64_t &presentationTimeUs)
+int32_t DemuxerCapiMock::GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+    const uint32_t index, uint64_t &relativePresentationTimeUs)
 {
     return AV_ERR_OK;
 }
diff --git a/test/unittest/demuxer_test/capi/demuxer_capi_mock.h b/test/unittest/demuxer_test/capi/demuxer_capi_mock.h
index 558b6feb8..3706b5c55 100644
--- a/test/unittest/demuxer_test/capi/demuxer_capi_mock.h
+++ b/test/unittest/demuxer_test/capi/demuxer_capi_mock.h
@@ -40,10 +40,10 @@ public:
     int32_t SetMediaKeySystemInfoCallback(bool isNull) override;
     int32_t SetDemuxerMediaKeySystemInfoCallback(bool isNull) override;
     int32_t GetMediaKeySystemInfo() override;
-    int32_t GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-        int64_t presentationTimeUs, uint32_t &frameIndex) override;
-    int32_t GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-        uint32_t frameIndex, int64_t &presentationTimeUs) override;
+    int32_t GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+        const uint64_t relativePresentationTimeUs, uint32_t &index) override;
+    int32_t GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+        const uint32_t index, uint64_t &relativePresentationTimeUs) override;
 private:
     OH_AVDemuxer *demuxer_ = nullptr;
 };
diff --git a/test/unittest/demuxer_test/demuxer_mock.h b/test/unittest/demuxer_test/demuxer_mock.h
index 302e3bccf..edfce2666 100644
--- a/test/unittest/demuxer_test/demuxer_mock.h
+++ b/test/unittest/demuxer_test/demuxer_mock.h
@@ -50,10 +50,10 @@ public:
     {
         return AV_ERR_OK;
     }
-    virtual int32_t GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-        int64_t presentationTimeUs, uint32_t &frameIndex) = 0;
-    virtual int32_t GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-        uint32_t frameIndex, int64_t &presentationTimeUs) = 0;
+    virtual int32_t GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+        const uint64_t relativePresentationTimeUs, uint32_t &index) = 0;
+    virtual int32_t GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+        const uint32_t index, uint64_t &relativePresentationTimeUs) = 0;
 };
 
 class __attribute__((visibility("default"))) AVDemuxerMockFactory {
diff --git a/test/unittest/demuxer_test/demuxer_unit_test.cpp b/test/unittest/demuxer_test/demuxer_unit_test.cpp
index 7cf37f610..dab6f9cbd 100644
--- a/test/unittest/demuxer_test/demuxer_unit_test.cpp
+++ b/test/unittest/demuxer_test/demuxer_unit_test.cpp
@@ -2261,74 +2261,74 @@ HWTEST_F(DemuxerUnitTest, Demuxer_GetMediaKeySystemInfo_4004, TestSize.Level1)
 
 #ifndef DEMUXER_CAPI_UNIT_TEST
 /**
- * @tc.name: Demuxer_GetFrameIndexByPresentationTimeUs_1000
- * @tc.desc: Get frameIndex by pts(audio track)
+ * @tc.name: Demuxer_GetIndexByRelativePresentationTimeUs_1000
+ * @tc.desc: Get index by pts(audio track)
  * @tc.type: FUNC
  */
-HWTEST_F(DemuxerUnitTest, Demuxer_GetFrameIndexByPresentationTimeUs_1000, TestSize.Level1)
+HWTEST_F(DemuxerUnitTest, Demuxer_GetIndexByRelativePresentationTimeUs_1000, TestSize.Level1)
 {
     InitResource(g_ptsConversionPath, LOCAL);
     ASSERT_NE(source_, nullptr);
     ASSERT_NE(demuxer_, nullptr);
 
     uint32_t trackIndex = 0;
-    int64_t presentationTimeUs = 69659;
-    uint32_t frameIndex = 0;
+    uint64_t relativePresentationTimeUs = 69659;
+    uint32_t index = 0;
 
-    int32_t ret = demuxer_->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
+    int32_t ret = demuxer_->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
     ASSERT_EQ(ret, AV_ERR_OK);
 
-    ASSERT_EQ(frameIndex, 3);
+    ASSERT_EQ(index, 3);
 }
 
 /**
- * @tc.name: Demuxer_GetFrameIndexByPresentationTimeUs_1001
- * @tc.desc: Get frameIndex by pts(video track)
+ * @tc.name: Demuxer_GetIndexByRelativePresentationTimeUs_1001
+ * @tc.desc: Get index by pts(video track)
  * @tc.type: FUNC
  */
-HWTEST_F(DemuxerUnitTest, Demuxer_GetFrameIndexByPresentationTimeUs_1001, TestSize.Level1)
+HWTEST_F(DemuxerUnitTest, Demuxer_GetIndexByRelativePresentationTimeUs_1001, TestSize.Level1)
 {
     InitResource(g_ptsConversionPath, LOCAL);
     ASSERT_NE(source_, nullptr);
     ASSERT_NE(demuxer_, nullptr);
 
     uint32_t trackIndex = 1;
-    int64_t presentationTimeUs = 66666;
-    uint32_t frameIndex = 0;
+    uint64_t relativePresentationTimeUs = 66666;
+    uint32_t index = 0;
 
-    int32_t ret = demuxer_->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
+    int32_t ret = demuxer_->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
     ASSERT_EQ(ret, AV_ERR_OK);
 
-    ASSERT_EQ(frameIndex, 4);
+    ASSERT_EQ(index, 4);
 }
 
 /**
- * @tc.name: Demuxer_GetFrameIndexByPresentationTimeUs_1002
- * @tc.desc: Get frameIndex by pts(not MP4)
+ * @tc.name: Demuxer_GetIndexByRelativePresentationTimeUs_1002
+ * @tc.desc: Get index by pts(not MP4)
  * @tc.type: FUNC
  */
-HWTEST_F(DemuxerUnitTest, Demuxer_GetFrameIndexByPresentationTimeUs_1002, TestSize.Level1)
+HWTEST_F(DemuxerUnitTest, Demuxer_GetIndexByRelativePresentationTimeUs_1002, TestSize.Level1)
 {
     InitResource(g_flvPath, LOCAL);
     ASSERT_NE(source_, nullptr);
     ASSERT_NE(demuxer_, nullptr);
 
     uint32_t trackIndex = 0;
-    int64_t presentationTimeUs = 69659;
-    uint32_t frameIndex = 0;
+    uint64_t relativePresentationTimeUs = 69659;
+    uint32_t index = 0;
 
-    int32_t ret = demuxer_->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
+    int32_t ret = demuxer_->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
     ASSERT_NE(ret, AV_ERR_OK);
 
-    ASSERT_EQ(frameIndex, 0);
+    ASSERT_EQ(index, 0);
 }
 
 /**
- * @tc.name: Demuxer_GetFrameIndexByPresentationTimeUs_1003
- * @tc.desc: Get frameIndex by pts(non-standard pts & different track)
+ * @tc.name: Demuxer_GetIndexByRelativePresentationTimeUs_1003
+ * @tc.desc: Get index by pts(non-standard pts & different track)
  * @tc.type: FUNC
  */
-HWTEST_F(DemuxerUnitTest, Demuxer_GetFrameIndexByPresentationTimeUs_1003, TestSize.Level1)
+HWTEST_F(DemuxerUnitTest, Demuxer_GetIndexByRelativePresentationTimeUs_1003, TestSize.Level1)
 {
     InitResource(g_ptsConversionPath, LOCAL);
     ASSERT_NE(source_, nullptr);
@@ -2336,89 +2336,89 @@ HWTEST_F(DemuxerUnitTest, Demuxer_GetFrameIndexByPresentationTimeUs_1003, TestSi
 
     // standard pts & video track
     uint32_t trackIndex = 1;
-    int64_t presentationTimeUs = 166666;
-    uint32_t frameIndex = 0;
-    int32_t ret = demuxer_->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
+    uint64_t relativePresentationTimeUs = 166666;
+    uint32_t index = 0;
+    int32_t ret = demuxer_->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
     ASSERT_EQ(ret, AV_ERR_OK);
-    ASSERT_EQ(frameIndex, 10);
+    ASSERT_EQ(index, 10);
 
     // non-standard pts & video track
     trackIndex = 1;
-    presentationTimeUs = 166667;
-    frameIndex = 0;
-    ret = demuxer_->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
+    relativePresentationTimeUs = 166667;
+    index = 0;
+    ret = demuxer_->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
     ASSERT_NE(ret, AV_ERR_OK);
 
     // standard pts & audio track
     trackIndex = 0;
-    presentationTimeUs = 232199;
-    frameIndex = 0;
-    ret = demuxer_->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
+    relativePresentationTimeUs = 232199;
+    index = 0;
+    ret = demuxer_->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
     ASSERT_EQ(ret, AV_ERR_OK);
-    ASSERT_EQ(frameIndex, 10);
+    ASSERT_EQ(index, 10);
 }
 
 /**
- * @tc.name: Demuxer_GetPresentationTimeUsByFrameIndex_1000
+ * @tc.name: Demuxer_GetRelativePresentationTimeUsByIndex_1000
  * @tc.desc: get pts by frameIndex(audio track)
  * @tc.type: FUNC
  */
-HWTEST_F(DemuxerUnitTest, Demuxer_GetPresentationTimeUsByFrameIndex_1000, TestSize.Level1)
+HWTEST_F(DemuxerUnitTest, Demuxer_GetRelativePresentationTimeUsByIndex_1000, TestSize.Level1)
 {
     InitResource(g_ptsConversionPath, LOCAL);
     ASSERT_NE(source_, nullptr);
     ASSERT_NE(demuxer_, nullptr);
 
     uint32_t trackIndex = 0;
-    int64_t presentationTimeUs = 0;
-    uint32_t frameIndex = 2;
+    uint64_t relativePresentationTimeUs = 0;
+    uint32_t index = 2;
 
-    int32_t ret = demuxer_->GetPresentationTimeUsByFrameIndex(trackIndex, frameIndex, presentationTimeUs);
+    int32_t ret = demuxer_->GetRelativePresentationTimeUsByIndex(trackIndex, index, relativePresentationTimeUs);
     ASSERT_EQ(ret, AV_ERR_OK);
 
-    ASSERT_EQ(presentationTimeUs, 46439);
+    ASSERT_EQ(relativePresentationTimeUs, 46439);
 }
 
 /**
- * @tc.name: Demuxer_GetPresentationTimeUsByFrameIndex_1001
+ * @tc.name: Demuxer_GetRelativePresentationTimeUsByIndex_1001
  * @tc.desc: get pts by frameIndex(video track)
  * @tc.type: FUNC
  */
-HWTEST_F(DemuxerUnitTest, Demuxer_GetPresentationTimeUsByFrameIndex_1001, TestSize.Level1)
+HWTEST_F(DemuxerUnitTest, Demuxer_GetRelativePresentationTimeUsByIndex_1001, TestSize.Level1)
 {
     InitResource(g_ptsConversionPath, LOCAL);
     ASSERT_NE(source_, nullptr);
     ASSERT_NE(demuxer_, nullptr);
 
     uint32_t trackIndex = 1;
-    int64_t presentationTimeUs = 0;
-    uint32_t frameIndex = 2;
+    uint64_t relativePresentationTimeUs = 0;
+    uint32_t index = 2;
 
-    int32_t ret = demuxer_->GetPresentationTimeUsByFrameIndex(trackIndex, frameIndex, presentationTimeUs);
+    int32_t ret = demuxer_->GetRelativePresentationTimeUsByIndex(trackIndex, index, relativePresentationTimeUs);
     ASSERT_EQ(ret, AV_ERR_OK);
 
-    ASSERT_EQ(presentationTimeUs, 33333);
+    ASSERT_EQ(relativePresentationTimeUs, 33333);
 }
 
 /**
- * @tc.name: Demuxer_GetPresentationTimeUsByFrameIndex_1002
+ * @tc.name: Demuxer_GetRelativePresentationTimeUsByIndex_1002
  * @tc.desc: get pts by frameIndex(not MP4)
  * @tc.type: FUNC
  */
-HWTEST_F(DemuxerUnitTest, Demuxer_GetPresentationTimeUsByFrameIndex_1002, TestSize.Level1)
+HWTEST_F(DemuxerUnitTest, Demuxer_GetRelativePresentationTimeUsByIndex_1002, TestSize.Level1)
 {
     InitResource(g_flvPath, LOCAL);
     ASSERT_NE(source_, nullptr);
     ASSERT_NE(demuxer_, nullptr);
 
     uint32_t trackIndex = 0;
-    int64_t presentationTimeUs = 0;
-    uint32_t frameIndex = 10;
+    uint64_t relativePresentationTimeUs = 0;
+    uint32_t index = 10;
 
-    int32_t ret = demuxer_->GetPresentationTimeUsByFrameIndex(trackIndex, frameIndex, presentationTimeUs);
+    int32_t ret = demuxer_->GetRelativePresentationTimeUsByIndex(trackIndex, index, relativePresentationTimeUs);
     ASSERT_NE(ret, AV_ERR_OK);
 
-    ASSERT_EQ(presentationTimeUs, 0);
+    ASSERT_EQ(relativePresentationTimeUs, 0);
 }
 
 /**
@@ -2433,18 +2433,18 @@ HWTEST_F(DemuxerUnitTest, Demuxer_PtsAndFrameIndexConversion_1000, TestSize.Leve
     ASSERT_NE(format_, nullptr);
 
     uint32_t trackIndex = 0;
-    int64_t presentationTimeUs = 92879;
-    uint32_t frameIndex = 0;
+    uint64_t relativePresentationTimeUs = 92879;
+    uint32_t index = 0;
 
-    int32_t ret = demuxer_->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
+    int32_t ret = demuxer_->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
     ASSERT_EQ(ret, AV_ERR_OK);
-    ASSERT_EQ(frameIndex, 4);
+    ASSERT_EQ(index, 4);
 
-    int64_t presentationTimeUs1 = 0;
+    uint64_t relativePresentationTimeUs1 = 0;
 
-    ret = demuxer_->GetPresentationTimeUsByFrameIndex(trackIndex, frameIndex, presentationTimeUs1);
+    ret = demuxer_->GetRelativePresentationTimeUsByIndex(trackIndex, index, relativePresentationTimeUs1);
     ASSERT_EQ(ret, AV_ERR_OK);
-    ASSERT_EQ(presentationTimeUs1, 92879);
+    ASSERT_EQ(relativePresentationTimeUs1, 92879);
 }
 
 /**
@@ -2459,18 +2459,18 @@ HWTEST_F(DemuxerUnitTest, Demuxer_PtsAndFrameIndexConversion_1001, TestSize.Leve
     ASSERT_NE(format_, nullptr);
 
     uint32_t trackIndex = 0;
-    int64_t presentationTimeUs = 0;
-    uint32_t frameIndex = 4;
+    uint64_t relativePresentationTimeUs = 0;
+    uint32_t index = 4;
 
-    int32_t ret = demuxer_->GetPresentationTimeUsByFrameIndex(trackIndex, frameIndex, presentationTimeUs);
+    int32_t ret = demuxer_->GetRelativePresentationTimeUsByIndex(trackIndex, index, relativePresentationTimeUs);
     ASSERT_EQ(ret, AV_ERR_OK);
-    ASSERT_EQ(presentationTimeUs, 92879);
+    ASSERT_EQ(relativePresentationTimeUs, 92879);
 
-    uint32_t frameIndex1 = 0;
+    uint32_t index1 = 0;
 
-    ret = demuxer_->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex1);
+    ret = demuxer_->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index1);
     ASSERT_EQ(ret, AV_ERR_OK);
-    ASSERT_EQ(frameIndex1, 4);
+    ASSERT_EQ(index1, 4);
 }
 #endif
 
diff --git a/test/unittest/demuxer_test/inner/demuxer_inner_buffer_mock.cpp b/test/unittest/demuxer_test/inner/demuxer_inner_buffer_mock.cpp
index 6bce8e9c9..b99779cee 100644
--- a/test/unittest/demuxer_test/inner/demuxer_inner_buffer_mock.cpp
+++ b/test/unittest/demuxer_test/inner/demuxer_inner_buffer_mock.cpp
@@ -98,20 +98,20 @@ int32_t DemuxerInnerMock::SeekToTime(int64_t mSeconds, SeekMode mode)
     return AV_ERR_UNKNOWN;
 }
 
-int32_t DemuxerInnerMock::GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-    int64_t presentationTimeUs, uint32_t &frameIndex)
+int32_t DemuxerInnerMock::GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+    const uint64_t relativePresentationTimeUs, uint32_t &index)
 {
     if (demuxer_ != nullptr) {
-        return demuxer_->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
+        return demuxer_->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
     }
     return AV_ERR_UNKNOWN;
 }
 
-int32_t DemuxerInnerMock::GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-    uint32_t frameIndex, int64_t &presentationTimeUs)
+int32_t DemuxerInnerMock::GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+    const uint32_t index, uint64_t &relativePresentationTimeUs)
 {
     if (demuxer_ != nullptr) {
-        return demuxer_->GetPresentationTimeUsByFrameIndex(trackIndex, frameIndex, presentationTimeUs);
+        return demuxer_->GetRelativePresentationTimeUsByIndex(trackIndex, index, relativePresentationTimeUs);
     }
     return AV_ERR_UNKNOWN;
 }
diff --git a/test/unittest/demuxer_test/inner/demuxer_inner_mock.cpp b/test/unittest/demuxer_test/inner/demuxer_inner_mock.cpp
index ba4b0ce47..47988317b 100644
--- a/test/unittest/demuxer_test/inner/demuxer_inner_mock.cpp
+++ b/test/unittest/demuxer_test/inner/demuxer_inner_mock.cpp
@@ -67,20 +67,21 @@ int32_t DemuxerInnerMock::SeekToTime(int64_t mSeconds, SeekMode mode)
     return AV_ERR_UNKNOWN;
 }
 
-int32_t DemuxerInnerMock::GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-    int64_t presentationTimeUs, uint32_t &frameIndex)
+int32_t DemuxerInnerMock::GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+    const uint64_t relativePresentationTimeUs, uint32_t &index)
 {
     if (demuxer_ != nullptr) {
-        return demuxer_->GetFrameIndexByPresentationTimeUs(trackIndex, presentationTimeUs, frameIndex);
+        return demuxer_->GetIndexByRelativePresentationTimeUs(trackIndex, relativePresentationTimeUs, index);
     }
     return AV_ERR_UNKNOWN;
 }
 
-int32_t DemuxerInnerMock::GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-    uint32_t frameIndex, int64_t &presentationTimeUs)
+int32_t DemuxerInnerMock::GetRelativePresentationTimeUsByIndex(uint32_t trackIndex,
+    const uint32_t index, uint64_t &relativePresentationTimeUs)
+
 {
     if (demuxer_ != nullptr) {
-        return demuxer_->GetPresentationTimeUsByFrameIndex(trackIndex, frameIndex, presentationTimeUs);
+        return demuxer_->GetRelativePresentationTimeUsByIndex(trackIndex, index, relativePresentationTimeUs);
     }
     return AV_ERR_UNKNOWN;
 }
diff --git a/test/unittest/demuxer_test/inner/demuxer_inner_mock.h b/test/unittest/demuxer_test/inner/demuxer_inner_mock.h
index 7d9623bbe..58679424d 100644
--- a/test/unittest/demuxer_test/inner/demuxer_inner_mock.h
+++ b/test/unittest/demuxer_test/inner/demuxer_inner_mock.h
@@ -33,10 +33,10 @@ public:
     int32_t ReadSample(uint32_t trackIndex, std::shared_ptr<AVMemoryMock> sample,
         AVCodecBufferInfo *bufferInfo, uint32_t &flag, bool checkBufferInfo) override;
     int32_t SeekToTime(int64_t mSeconds, SeekMode mode) override;
-    int32_t GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-        int64_t presentationTimeUs, uint32_t &frameIndex) override;
-    int32_t GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-        uint32_t frameIndex, int64_t &presentationTimeUs) override;
+    int32_t GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+        const uint64_t relativePresentationTimeUs, uint32_t &index) override;
+    int32_t GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+        const uint32_t index, uint64_t &relativePresentationTimeUs) override;
 
 private:
     std::shared_ptr<AVDemuxer> demuxer_ = nullptr;
diff --git a/test/unittest/media_demuxer_test/media_demuxer_unit_test.cpp b/test/unittest/media_demuxer_test/media_demuxer_unit_test.cpp
index d68a2d635..685feb617 100644
--- a/test/unittest/media_demuxer_test/media_demuxer_unit_test.cpp
+++ b/test/unittest/media_demuxer_test/media_demuxer_unit_test.cpp
@@ -888,17 +888,17 @@ HWTEST_F(MediaDemuxerUnitTest, MediaDemuxer_RegisterVideoStreamReadyCallback_010
 }
 
 
-HWTEST_F(MediaDemuxerUnitTest, MediaDemuxer_GetPresentationTimeUsByFrameIndex_010, TestSize.Level1)
+HWTEST_F(MediaDemuxerUnitTest, MediaDemuxer_GetRelativePresentationTimeUsByIndex_010, TestSize.Level1)
 {
     std::shared_ptr<MediaDemuxer> demuxer = std::make_shared<MediaDemuxer>();
-    int64_t presentationTimeUs;
-    EXPECT_EQ(demuxer->GetPresentationTimeUsByFrameIndex(0, 1, presentationTimeUs), Status::ERROR_NULL_POINTER);
-    uint32_t frameIndex;
-    EXPECT_EQ(demuxer->GetFrameIndexByPresentationTimeUs(0, 1, frameIndex), Status::ERROR_NULL_POINTER);
+    uint64_t relativePresentationTimeUs;
+    EXPECT_EQ(demuxer->GetRelativePresentationTimeUsByIndex(0, 1, relativePresentationTimeUs), Status::ERROR_NULL_POINTER);
+    uint32_t index;
+    EXPECT_EQ(demuxer->GetIndexByRelativePresentationTimeUs(0, 1, index), Status::ERROR_NULL_POINTER);
 
     demuxer->demuxerPluginManager_ = nullptr;
-    EXPECT_EQ(demuxer->GetPresentationTimeUsByFrameIndex(0, 1, presentationTimeUs), Status::ERROR_NULL_POINTER);
-    EXPECT_EQ(demuxer->GetFrameIndexByPresentationTimeUs(0, 1, frameIndex), Status::ERROR_NULL_POINTER);
+    EXPECT_EQ(demuxer->GetRelativePresentationTimeUsByIndex(0, 1, relativePresentationTimeUs), Status::ERROR_NULL_POINTER);
+    EXPECT_EQ(demuxer->GetIndexByRelativePresentationTimeUs(0, 1, index), Status::ERROR_NULL_POINTER);
 }
 
 HWTEST_F(MediaDemuxerUnitTest, MediaDemuxer_CheckDropAudioFrame_013, TestSize.Level1)
@@ -1018,8 +1018,8 @@ HWTEST_F(MediaDemuxerUnitTest, MediaDemuxer_GetPresentation_001, TestSize.Level1
         AVBufferQueue::Create(8, MemoryType::SHARED_MEMORY, "testInputBufferQueue");
     sptr<AVBufferQueueProducer> inputBufferQueueProducer = inputBufferQueue->GetProducer();
     EXPECT_EQ(demuxer->SetOutputBufferQueue(0, inputBufferQueueProducer), Status::OK);
-    int64_t presentationTimeUs = 0;
-    EXPECT_EQ(Status::OK, demuxer->GetPresentationTimeUsByFrameIndex(0, 0, presentationTimeUs));
+    uint64_t relativePresentationTimeUs = 0;
+    EXPECT_EQ(Status::OK, demuxer->GetRelativePresentationTimeUsByIndex(0, 0, relativePresentationTimeUs));
 }
 
 HWTEST_F(MediaDemuxerUnitTest, MediaDemuxer_GetFrameIndex_001, TestSize.Level1)
@@ -1041,8 +1041,8 @@ HWTEST_F(MediaDemuxerUnitTest, MediaDemuxer_GetFrameIndex_001, TestSize.Level1)
     sptr<AVBufferQueueProducer> inputBufferQueueProducer = inputBufferQueue->GetProducer();
     EXPECT_EQ(demuxer->SetOutputBufferQueue(0, inputBufferQueueProducer), Status::OK);
     sleep(1);
-    uint32_t frameIndex = 0;
-    EXPECT_EQ(Status::ERROR_MISMATCHED_TYPE, demuxer->GetFrameIndexByPresentationTimeUs(0, 0, frameIndex));
+    uint32_t index = 0;
+    EXPECT_EQ(Status::ERROR_MISMATCHED_TYPE, demuxer->GetIndexByRelativePresentationTimeUs(0, 0, index));
 }
 
 HWTEST_F(MediaDemuxerUnitTest, MediaDemuxer_CheckDropAudioFrame_001, TestSize.Level1)
diff --git a/test/unittest/media_demuxer_test/media_demuxer_unit_test.h b/test/unittest/media_demuxer_test/media_demuxer_unit_test.h
index 0eeaf0e6f..53be048d5 100644
--- a/test/unittest/media_demuxer_test/media_demuxer_unit_test.h
+++ b/test/unittest/media_demuxer_test/media_demuxer_unit_test.h
@@ -135,13 +135,13 @@ public:
     {
         return mapStatus_[name_];
     }
-    virtual Status GetFrameIndexByPresentationTimeUs(uint32_t trackIndex,
-        int64_t presentationTimeUs, uint32_t &frameIndex)
+    virtual Status GetIndexByRelativePresentationTimeUs(const uint32_t trackIndex,
+        const uint64_t relativePresentationTimeUs, uint32_t &index)
     {
         return mapStatus_[name_];
     }
-    virtual  Status GetPresentationTimeUsByFrameIndex(uint32_t trackIndex,
-        uint32_t frameIndex, int64_t &presentationTimeUs)
+    virtual  Status GetRelativePresentationTimeUsByIndex(const uint32_t trackIndex,
+        const uint32_t index, uint64_t &relativePresentationTimeUs)
     {
         return mapStatus_[name_];
     }
